04.5-rmarkdown.Rmd

```diff
@@ -0,0 +1,415 @@
+```
+
+This chunk also has a name, "cars". It is not necessary to name chunks, but it can help you to navigate to a particular chunk later, if you do name them.
+
+### Running chunks
+
+Here we will see a green triangular button with its point facing to the right. This is the play button. If you try pressing this button inside of RStudio, you will see a preview of what the code does. It should show the summary of the `cars` data.
+
+
+
+
+```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "An image of what the code looks like after the play button is pressed for the second chunk."}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g229ab7a949e_0_292")
+```
+
+Pretty nifty!
+
+This is similar to testing our code in the console, in that if we assign an object it will show up in the environment.
+
+```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Running a code chunk is like testing in the console, it will add any objects to the environment. Image shows that pressing the run button on a chunk that assigns data will create an object that shows up in the environment."}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g229ab7a949e_0_311")
+```
+
+<div class = "dictionary">
+
+- **Code chunk** - A piece of code in an R Markdown file. The code can be previewed pressing the play button for the chunk, which is equivalent to running the code in the console.
+
+</div>
+
+
+### Running previous chunks
+
+You may also notice that there is another button to the left of the play button. This button allows you to play all previous chunks before this chunk.
+
+```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "An image of the second R chunk in a new R Markdown file. Importantly there is a play previous button that allows you to run the code within the chunks before this chunk."}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g229ab7a949e_0_297")
+```
+
+This is super helpful for making sure that your code works properly and isn't working simply because you have something in your environment from code that you tested in the console but did not save it anywhere. Other issues can happen if you run a code chunk out of order or changed the code in the chunk after running it previously. Therefore we recommend cleaning the environment (which we will describe in the next section) and testing out that your chunks work well together.
+
+## Cleaning the environment
+
+We also suggest cleaning out your environment somewhat regularly when you are interactively testing your R Markdown file using chunks. To do so, you can press the button that looks like a broom in the upper right pane. 
```

Comment:
Restarting R Session's too is important, because Cleaning out the R environment won't clear out packages which is also important for reproducible. I don't know what the appropriate time is to tell them this tidbit though. Might not be here but it might be.

Reply:
hmmm maybe in your chapter about package versions?

Reply:
I could also add it here...